server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1646,30 +1717,26 @@ public boolean empty() {
 
                 @Override
                 public boolean advanceToDoc(int docId) throws IOException {
-                    return hasValue = leaf.advanceExact(docId);
+                    idx = Arrays.binarySearch(docIdsInLeaf, docId);
```

Comment:
Do we need to do a binary search here, given that we know we're visiting the docs from `docIdsInLeaf` in-order?

Reply:
Are you saying everytime folks call this we just advance `idx` and check? I think that's fine, yes.

Reply:
Yes, exactly.  Won't make a big difference for small sets of documents, but with `size=1000` and lots of fields then I think avoiding the binary search is a win.